pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/PancanAddAdditionalDiagnosis.java

```diff
@@ -0,0 +1,294 @@
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.google.gson.Gson;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigException;
+import com.typesafe.config.ConfigFactory;
+import com.typesafe.config.ConfigResolveOptions;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.broadinstitute.ddp.db.dao.JdbiPicklistOption;
+import org.broadinstitute.ddp.db.dao.PicklistQuestionDao;
+import org.broadinstitute.ddp.db.dto.PicklistGroupDto;
+import org.broadinstitute.ddp.db.dto.PicklistOptionDto;
+import org.broadinstitute.ddp.exception.DDPException;
+import org.broadinstitute.ddp.model.activity.definition.question.PicklistOptionDef;
+import org.broadinstitute.ddp.util.ConfigUtil;
+import org.broadinstitute.ddp.util.GsonUtil;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.sqlobject.SqlObject;
+import org.jdbi.v3.sqlobject.config.RegisterConstructorMapper;
+import org.jdbi.v3.sqlobject.customizer.Bind;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+import org.jdbi.v3.sqlobject.statement.SqlUpdate;
+
+@Slf4j
+public final class PancanAddAdditionalDiagnosis implements CustomTask {
+    private static final String PATCH_CONF_NAME = "add-additional-diagnosis.conf";
+
+    @AllArgsConstructor(access = AccessLevel.PRIVATE)
+    private static enum Metadata {
+        DESCRIPTION("description"),
+        STUDY("study"),
+        EXPECTED_OPTIONS("expected-options"),
+        TARGET_GROUP_SID("target-group-sid"),
+        INSERT_AFTER_SID("insert-after-sid"),
+        NEW_OPTIONS("options");
+
+        @Getter
+        private final String key;
+    }
+
+    private static interface SqlHelper extends SqlObject {
+        
+        @SqlQuery("SELECT"
+                + "    pg.picklist_group_id AS picklist_group_id,"
+                + "    pg.group_stable_id AS group_stable_id,"
+                + "    pg.name_template_id AS name_template_id,"
+                + "    pg.display_order AS display_order,"
+                + "    pg.revision_id AS revision_id,"
+                + "    revision.start_date AS revision_start_timestamp,"
+                + "    revision.end_date AS revision_end_timestamp"
+                + "  FROM picklist_group AS pg"
+                + "  JOIN picklist_question AS pq ON pq.question_id = pg.picklist_question_id"
+                + "  JOIN question AS q ON q.question_id = pq.question_id"
+                + "  JOIN study_activity AS sa ON q.study_activity_id = sa.study_activity_id"
+                + "  JOIN umbrella_study AS us ON sa.study_id = us.umbrella_study_id"
+                + "  JOIN revision ON revision.revision_id = pg.revision_id"
+                + "  WHERE group_stable_id = :groupStableId"
+                + "    AND us.guid = :studyGuid")
+        @RegisterConstructorMapper(PicklistGroupDto.class)
+        public List<PicklistGroupDto> fetchPicklistGroupsByName(
```

Comment:
Interface methods are public by default, so these visibility specifiers are unnecessary. Leaving them for now to prevent dismissing existing reviews on the PR.